experimental/origin-dapp2/src/pages/nav/Messages.js

```diff
@@ -56,17 +70,68 @@ class MessagesNav extends Component {
   }
 }
 
-const MessagesDropdown = ({ onClick }) => {
+const MessagesDropdown = props => {
+  const { onClick, wallet } = props
+
   return (
     <div className="dropdown-menu dropdown-menu-right show messages">
-      <Link to="/messages" onClick={() => onClick()}>
-        View Messages
-      </Link>
+      <Query query={ConversationsQuery} pollInterval={2000}>
+        {({ data, error, loading }) => {
+          if (loading || error) return null
+          const conversations = get(data, 'messaging.conversations', [])
+          const messages = conversations.map(({ messages }) => messages)
+          const totalUnreadMessages = messages.flat().reduce((result, msg) => {
```

Comment:
I think this can be a field on GraphQL. In general we should try to offload logic like this to GraphQL

Reply:
Perhaps we can pass in the user's wallet as a variable to the Query. There's an example of doing that on the Room component

Reply:
Right, of course. Thanks